datadog_checks_dev/datadog_checks/dev/config_validator/config_block.py

```diff
@@ -0,0 +1,452 @@
+COMMENT_REGEX = "^ {INDENT}##(.*)$"
+
+# Regex to match any comment with no respect to indentation
+INCORRECTLY_INDENTED_COMMENT_REGEX = "^ *##(.*)$"
+
+
+class ParamProperties:
+    """Class to represent a parameter declared using the '@param' annotation
+    """
+
+    def __init__(self, var_name, type_name, required=True, default_value=None):
+        self.var_name = var_name
+        self.type_name = type_name
+        self.required = required
+        self.default_value = default_value
+
+    @staticmethod
+    def parse_from_string(idx, config_lines, indent, errors=None):
+        if errors is None:
+            errors = []
+
+        if not is_exactly_indented(config_lines[idx], indent):
+            errors.append(ValidatorError("Content is not correctly indented", idx))
+            return None
+
+        current_line = config_lines[idx][indent:]
+
+        if not current_line.startswith("## @param"):
+            errors.append(ValidatorError("Expecting @param declaration", idx))
+            return None
+
+        m = re.match(PARAM_REGEX, current_line)
+        if m is None:
+            errors.append(ValidatorError("Invalid @param declaration", idx))
+            return None
+
+        if m.group(3) == "optional":
+            def_val = None
+            if m.group(4):  # If there is a default value
+                def_val = m.group(4)[12:]
+
+            return ParamProperties(m.group(1), m.group(2), False, def_val)
+
+        return ParamProperties(m.group(1), m.group(2), True)
+
+
+class ConfigBlock:
+    """Class to represent a 'configuration block' which is the definition of a variable, with the @param annotation,
+    its description and its content.
+    """
+
+    def __init__(self, param_prop, description, line, length, errors=None, should_recurse=False):
+        """
+        :param param_prop: ParamProperties instance
+        :param description: The description preceding the variable declaration
+        :param line: The first line of this block
+        :param length: The number of lines this block takes. (see should_recurse)
+        :param errors: An array of ValidatorError instances
+        :param should_recurse: Whether or not the content of the block must be analyzed. See _should_recurse function
+        """
+        self.param_prop = param_prop
+        self.description = description
+        self.line = line
+        self.length = length
+        if errors is None:
+            self.errors = []
+        self.should_recurse = should_recurse
+
+    def validate(self):
+        """Method to return a list of errors and warnings about an already parsed block"""
+        errors = []
+        errors += self._validate_description()
+        errors += self._validate_type()
+        return errors
+
+    def _validate_description(self):
+        """Check if the block has a description and lines are not too long."""
+        errors = []
+        if self.description is None:
+            return errors
```

Comment:
Don't you want to have an error here?

Reply:
description being None means that it could not be parsed correctly. In that case, an error already happened before that point. Validation of the description is skipped in that case.

Adding a comment on that.